single_kernel_mongo/events/backups.py

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Manager for handling backup events."""
+
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+from ops import MaintenanceStatus
+from ops.charm import ActionEvent, RelationJoinedEvent
+from ops.framework import Object
+
+from single_kernel_mongo.config.relations import ExternalRequirerRelations
+from single_kernel_mongo.exceptions import (
+    InvalidArgumentForActionError,
+    InvalidPBMStatusError,
+    ListBackupError,
+    PBMBusyError,
+    RestoreError,
+    ResyncError,
+    SetPBMConfigError,
+    WorkloadExecError,
+    WorkloadServiceError,
+)
+from single_kernel_mongo.lib.charms.data_platform_libs.v0.s3 import (
+    CredentialsChangedEvent,
+    S3Requirer,
+)
+from single_kernel_mongo.utils.event_helpers import (
+    defer_event_with_info_log,
+    fail_action_with_error_log,
+    success_action_with_info_log,
+)
+
+if TYPE_CHECKING:
+    from single_kernel_mongo.abstract_charm import AbstractMongoCharm
+    from single_kernel_mongo.managers.mongodb_operator import MongoDBOperator
+
+
+logger = logging.getLogger(__name__)
+
+INVALID_S3_INTEGRATION_STATUS = (
+    "Relation to s3-integrator is not supported, config role must be config-server."
+)
```

Comment:
It's a part I haven't worked on yet (as I said in the PR description).
I think it needs to be done and will be done in the next pulse, I pushed away the `status` managing part of the code to later because it needed some complex decision and discussions with you and I wanted to first have something running.
So there's a ticket for status management which will need a bit of design but this is later :)

Reply:
no worries! I can do it when I do statuses